dependency/vault_write.go

```diff
@@ -168,12 +170,20 @@ func (d *VaultWriteQuery) writeSecret(clients *ClientSet, opts *QueryOptions) (*
 		RawQuery: opts.String(),
 	})
 
-	vaultSecret, err := clients.Vault().Logical().Write(d.path, d.data)
+	data := d.data
+
+	_, isv2, _ := isKVv2(clients.Vault(), d.path)
+	if isv2 {
+		data = map[string]interface{}{"data": d.data}
+	}
+
+	vaultSecret, err := clients.Vault().Logical().Write(d.path, data)
 	if err != nil {
 		return nil, errors.Wrap(err, d.String())
 	}
+	// always nil when KVv1 engine, when isv2==false
 	if vaultSecret == nil {
```

Comment:
why not combine these with `if vaultSecret == nil && isv2`?

Reply:
Because I missed that refactor. Thanks.